dbms/src/Interpreters/Join.cpp

```diff
@@ -589,10 +722,28 @@ std::unique_ptr<IColumn::Offsets> NO_INLINE joinRightIndexedColumns(
 
             if (find_result.isFound())
             {
-                filter[i] = 1;
                 auto & mapped = find_result.getMapped();
-                mapped.setUsed();
-                addFoundRow<STRICTNESS, Map>(mapped, added_columns, current_offset);
+
+                if constexpr (STRICTNESS == ASTTableJoin::Strictness::Asof)
```

Comment:
It looks that it's possible to specialize addFoundRow with Asof instead of constexpr

Reply:
The issue here is twofold:
1. C++ does not support partial specialisation of a function template (so it would have to be a partially specialised struct with a static member function addFoundRow).
2. The addFoundRow function for asof takes an additional argument, which is the value for the asof column, as the KeyGetter only results in the value for the columns that were joined on exactly and not the asof column value. Therefore (without modifying the addFoundRow functions for Any and All to also take this argument), not using if constexpr does not seem to work.

Reply:
OK :)